org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandlerChainTest.java

```diff
@@ -267,4 +332,48 @@ public Object newObject() {
 		completionItems = list.getItems().stream().filter(i -> i.getLabel().contains("newObject")).collect(Collectors.toList());
 		assertEquals("emptyList completion count [type]", 0, completionItems.size());
 	}
+
+	@Test
+	public void testChainCompletionsWithToken_ExpectReplaceToken() throws Exception {
+		//@formatter:off
+			ICompilationUnit unit = getWorkingCopy(
+					"src/java/Foo.java",
+					"""
+						import java.util.List;
+						public class Foo {
+						    public static void main(String[] args) {
+								List<String> names = empty
+						    }
+						}
+						""");
+		//@formatter:on
+		CompletionList list = requestCompletions(unit, "names = empty");
+		List<CompletionItem> completionItems = list.getItems().stream().filter(i -> i.getLabel().contains("emptyList")).collect(Collectors.toList());
+		assertEquals("emptyList completion count", 1, completionItems.size());
+
+		CompletionItem completionItem = completionItems.get(0);
+		assertNotNull(completionItem);
+		assertEquals("Completion getTextEditText", "Collections.emptyList()", completionItem.getTextEdit().getLeft().getNewText());
+		var loc = findCompletionLocation(unit, "names = empty", 0);
+		var expected = new Position(loc[0], loc[1] - "empty".length());
+		assertEquals("Completion TextEdit Range", expected, completionItem.getTextEdit().getLeft().getRange().getStart());
+	}
+
+	@Test
+	public void testChainCompletionsOnVariableWithToken() throws Exception {
+		//@formatter:off
+			ICompilationUnit unit = getWorkingCopy(
+					"src/java/Foo.java",
+					"""
+						import java.util.Collection;
+						public class Foo {
+						    public static void main(String[] args) {
+								Collection<String> names = emptyL
+						    }
+						}
+						""");
+		//@formatter:on
+		CompletionList list = requestCompletions(unit, "names = emptyL");
+		assertTrue("all start with token", list.getItems().stream().allMatch(i -> i.getLabel().matches(".*\\.emptyList()")));
```

Comment:
This testcase failed when I tried it out as a real example. I never saw any `emptyList()` being suggested. When I dug closer I noticed for `allMatch(..)`:

```
/**
     * Returns whether all elements of this stream match the provided predicate.
     * May not evaluate the predicate on all elements if not necessary for
     * determining the result.  If the stream is empty then {@code true} is
     * returned and the predicate is not evaluated.
```

So an additional non-zero list check is needed.